### DISCUSSION TITLE
Unable to use Bitwarden CLI in code-server terminal

### DISCUSSION THREAD

**Discussion by exodar** [coder/code-server#4922] [Q&A] [answered]
When I execute the Bitwarden CLI from a code-server terminal, it immediately returns a prompt with no output whatsoever.  You can't even access --help.  However, if I SSH into the same box that code-server is running on with the same user that code-server is running as, I can execute it just fine.

I'm using the latest Bitwarden CLI from here:

https://vault.bitwarden.com/download/?app=cli&platform=linux

**Comment by lg-nineoneninethreenine** [comment]
Firstly - do you use HTTPS? A lot of these issues are related to WebView. Does Bitwarden really work locally v/s your code-server web env?

**Comment by exodar** [comment]
No I'm using HTTP to access my code-server.  Bitwarden does indeed work from an SSH terminal outside of VS Code, but nothing when using the code-server terminal.  This can be reproduced by downloading the Bitwarden CLI and just running 'bw --help'.

It's very strange.

**Reply by lg-nineoneninethreenine** [reply]
I sus WebView. Can you try enabling HTTPS on your instance?

**Comment by jsjoeio** [comment]
> However, if I SSH into the same box that code-server is running on with the same user that code-server is running as, I can execute it just fine.

I can also do this (running code-server on my macOS locally).

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/3806031/156061054-06ab79c3-7250-4465-b62c-4b246c19b96a.png">

I don't know why it wouldn't work on a remote machine though 🤔

Can you run `code-server --log trace` and then try using `bw` again? I wonder if we'll get anything in the server logs. Or even the browser logs.

**Comment by exodar** [comment]
Nothing appeared in trace log.  Also attached output of me running it directly in SSH session.
![Screen Shot 2022-02-28 at 4 40 53 PM](https://user-images.githubusercontent.com/6360493/156063768-b9784776-b0a8-4eef-be58-98e7e047680f.png)
![Screen Shot 2022-02-28 at 4 41 09 PM](https://user-images.githubusercontent.com/6360493/156063778-41f13fa8-ea62-4e79-9a3e-e9488682216b.png)
![Screen Shot 2022-02-28 at 4 43 59 PM](https://user-images.githubusercontent.com/6360493/156063824-2ed030e6-5dff-4cfd-870f-48cccdfe676d.png)

**Reply by jsjoeio** [reply]
This is very very odd. I do see "profile not validated". Is there a different shell profile you can try?

**Comment by exodar** [comment]
I'm using ZSH, but I've tried .bashrc and .profile as well as switching to /bin/bash.  Same results.  Assuming that's what you mean.

**Reply by jsjoeio** [reply]
Yes, that's what I mean :( @code-asher any ideas?

(side note: might be easier to keep this in one thread. If you reply to my direct message, it will keep it in one thread).

**Reply by jsjoeio** [reply]
![exampleee](https://user-images.githubusercontent.com/3806031/156203550-e05c8a1f-9735-4aef-b492-9c80b67d3e29.jpg)

**Reply by lg-nineoneninethreenine** [reply]
Woah @jsjoeio are you ok? I have never seen you posting screenshots like that! Everything good?

**Reply by jsjoeio** [reply]
LOL yes, don't worry. I made that in Preview on macOS 🤣 The replies in GitHub Discussions are confusing so hoping to make it more clear.

**Reply by code-asher** [reply]
Weird, I get the same issue.  I spun up code-server 4.1.0 and tried:

```
$ wget -O bw.zip 'https://vault.bitwarden.com/download/?app=cli&platform=linux'
$ unzip bw.zip
$ chmod +x bw
$ ./bw --help
```

**Reply by code-asher** [reply]
In my case this works:

```
NODE_OPTIONS= ./bw --help
```

It does not seem to like when `NODE_OPTIONS` is set.

We should remove this anyway: https://github.com/coder/code-server/issues/2878

But still not sure why this causes this issue.

**Comment by exodar** [comment]
Im hoping to try out HTTPS later this week when I have more time to configure it to see if that helps.  If it does help, I'm using Ubuntu Server 20.04 on an Intel platform for my code-server implementation.  There's nothing else installed on the box, it's super bare-bones.

**Reply by jsjoeio** [reply]
Okay let us know how that goes. 🤞

**Comment by exodar** [comment]
I’ll try this and report back. 

Thanks,

Rob Emenaker
***@***.***

**Comment by exodar** [comment]
This works!

**Comment by jsjoeio** [comment, marked as answer]
Solution posted by @code-asher 

> In my case this works:
> 
> ```
> NODE_OPTIONS= ./bw --help
> ```
> 
> It does not seem to like when `NODE_OPTIONS` is set.
> 
> We should remove this anyway: #2878
> 
> But still not sure why this causes this issue.
